components/count-badge/count-badge-mixin.js

```diff
@@ -0,0 +1,189 @@
+			 * @type {boolean}
+			 */
+			announceChanges: {
+				type: Boolean,
+				attribute: 'announce-changes'
+			},
+			/**
+			 * Optionally add a tooltip on the badge. Defaults to false.
+			 * @type {boolean}
+			 */
+			hasTooltip: {
+				type: Boolean,
+				attribute: 'has-tooltip'
+			},
+			/**
+			 * Optionally choose to not render the count badge when the number is zero. Defaults to false.
+			 * @type {boolean}
+			 */
+			hideZero: {
+				type: Boolean,
+				attribute: 'hide-zero'
+			},
+			/**
+			 * Optionally specify a digit limit, after which numbers are truncated. Defaults to two for "notification" type and no limit for "count" type.
+			 * @type {number}
+			 */
+			maxDigits: {
+				type: Number,
+				attribute: 'max-digits'
+			},
+			/**
+			 * The number to be displayed on the badge. Must be a positive integer.
+			 * @type {number}
+			 */
+			number: {
+				type: Number,
+				attribute: 'number'
+			},
+			/**
+			 * The size of the badge. Defaults to "small".
+			 * @type {'small'|'large'}
+			 */
+			size: {
+				type: String,
+				reflect: true,
+				attribute: 'size'
+			},
+			/**
+			 * Optionally choose to add a tab stop to the badge. Defaults to false.
+			 * @type {boolean}
+			 */
+			tabStop: {
+				type: Boolean,
+				attribute: 'tab-stop'
+			},
+			/**
+			 * * Descriptive text for the badge which will act as an accessible label and tooltip text when tooltips are enabled.
+			 * @type {string}
+			 */
+			text: {
+				type: String
+			},
+			/**
+			 * The type of the badge. Defaults to "count".
+			 * @type {'count'|'notification'}
+			 */
+			type: {
+				type: String,
+				reflect: true,
+				attribute: 'type'
+			}
+		};
+	}
+
+	constructor() {
+		super();
+		this.announceChanges = false;
+		this.hasTooltip = false;
+		this.hideZero = false;
+		this.size = 'small';
+		this.tabStop = false;
+		this.text = '';
+		this.type = 'count';
+		this._textId = getUniqueId();
+	}
+
+	connectedCallback() {
+		super.connectedCallback();
+		if (!this.maxDigits && this.type === 'notification') {
+			// default to two digits for notification type
+			this.maxDigits = 2;
+		}
+	}
+
+	getOffscreenId() {
+		return this.hasTooltip ? undefined : this._textId;
+	}
+
+	renderCount(numberStyles) {
+		let numberString = `${this.number}`;
+		const hideNumber = this.hideZero && this.number === 0;
+		if (hideNumber) {
+			numberString = '';
+		}
+		numberStyles = {
+			...numberStyles,
+			visibility: hideNumber ? 'hidden' : 'visible'
+		};
+		if (this.maxDigits && this.number.toString().length > this.maxDigits) {
+			numberString = `${'9'.repeat(this.maxDigits)}`;
+			numberString = formatNumber(parseNumber(numberString));
+			numberString = this.localize('components.count-badge.plus', { number: numberString });
+		} else if (!hideNumber) {
+			numberString = formatNumber(parseNumber(numberString));
```

Comment:
And then in this case the number is coming from the property directly, so it's already `Number` and shouldn't need parsing.
```suggestion
			numberString = formatNumber(numberString);
```